conda/gateways/repodata/__init__.py

```diff
@@ -630,7 +630,7 @@ def replace(self, temp_path: Path, target=None):
             try:
                 temp_path.rename(target)
             except FileExistsError:  # Windows
-                self.cache_path_json.unlink()
+                target.unlink()
```

Comment:
In L631 we try to move `temp_path` onto `target`, not `cache_path_json`; so we must unlink `target` if that fails.
An older version of this function only used `cache_path_json` and didn't have `target`.